src/managed_scip.jl

```diff
@@ -0,0 +1,74 @@
+# holds pointers to SCIP data and takes care of memory mgmt
+mutable struct ManagedSCIP
+    scip::Ref{Ptr{SCIP_}}
```

Comment:
What's the reason to use `Ref{Ptr{SCIP_}}` instead of `Ptr{SCIP_}` directly? (Edit: I see how it's used in `free_scip`. Still, the indirection might be removed for vars and conss.)

Reply:
Actually no, we do need `SCIP_VAR**` for `SCIPreleaseVar`, so I don't think we can do anything simpler.

See also [discussion on discourse](https://discourse.julialang.org/t/confusion-about-ptr-ref-and-dereferencing/18805).